### PR TITLE
better cache reuse

### PR DESCRIPTION
**Pull request by dtrawins**
### 🛠 Summary

CVS-148688
Optimize docker build to improve cache reusing in CI builds

### 🧪 Checklist

- [ ] Unit tests added.
- [ ] The documentation updated.
- [ ] Change follows security best practices.
``

### CODE REVIEW

**Review by rasapala — APPROVED** [review]

File: third_party/llm_engine/llm_engine.bzl [hunk 1]
```diff
@@ -14,6 +14,7 @@
 # limitations under the License.
 #
 
+
```
Comment (mzegla) [inline third_party/llm_engine/llm_engine.bzl:17]: Why additional new line? Is it linter/style-related?

File: Dockerfile.redhat [hunk 2]
```diff
@@ -44,8 +44,7 @@ RUN dnf install -y https://dl.fedoraproject.org/pub/epel/epel-release-latest-8.n
             patch \
             pkg-config \
             wget \
-            yum-utils \
```
Comment (dkalinowski) [inline Dockerfile.redhat:47]: We no longer need tbb in build image? How do we get one?

File: Dockerfile.redhat [hunk 3]
```diff
@@ -303,7 +303,7 @@ ARG minitrace_flags
 # Test Coverage
 COPY ci/check_coverage.bat /ovms/
 ARG CHECK_COVERAGE=0
-ARG RUN_TESTS=1
+ARG RUN_TESTS=0
```
Comment (dkalinowski) [inline Dockerfile.redhat:306]: Doesnt it disable unit tests by default? On-commit? Recompile master? RCs?

File: third_party/llm_engine/llm_engine.bzl (hunk 1 above)
Comment (mzegla) [reply third_party/llm_engine/llm_engine.bzl:17]:
```suggestion

```

**Review by dkalinowski — APPROVED** [review]

**Pull request merged**